src/forms/FormControl.scss

```diff
@@ -292,91 +513,134 @@
-        stroke: var(--color-accent-fg);
-      }
+    background: var(--color-accent-fg);
+    border-color: var(--color-accent-fg);
+    transition: background-color, border-color 80ms cubic-bezier(0.32, 0, 0.67, 0) 0ms; // unchecked -> checked
 
-      .FormControl--checkbox-check {
-        fill: var(--color-fg-on-emphasis);
-        visibility: visible;
+    &::before {
+      visibility: visible;
+      transition: visibility 0s linear 0s;
 
-        @media screen and (prefers-reduced-motion: no-preference) {
-          animation: checkmarkIn 200ms cubic-bezier(0.11, 0, 0.5, 0) forwards;
-        }
+      @media screen and (prefers-reduced-motion: no-preference) {
+        animation: checkmarkIn 80ms cubic-bezier(0.65, 0, 0.35, 1) forwards 80ms;
       }
     }
   }
 
-  &:indeterminate {
-    + .FormControl--checkbox-svg {
-      .FormControl--checkbox-background {
-        fill: var(--color-accent-fg);
-        stroke: var(--color-accent-fg);
-      }
+  &:focus,
+  &:focus-visible {
+    outline-offset: 2px;
+  }
 
-      .FormControl--checkbox-indeterminate {
-        fill: var(--color-fg-on-emphasis);
-        visibility: visible;
-      }
+  &:indeterminate {
+    &::before {
+      mask-image: url('data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAiIGhlaWdodD0iMiIgdmlld0JveD0iMCAwIDEwIDIiIGZpbGw9Im5vbmUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CjxwYXRoIGZpbGwtcnVsZT0iZXZlbm9kZCIgY2xpcC1ydWxlPSJldmVub2RkIiBkPSJNMCAxQzAgMC40NDc3MTUgMC40NDc3MTUgMCAxIDBIOUM5LjU1MjI5IDAgMTAgMC40NDc3MTUgMTAgMUMxMCAxLjU1MjI4IDkuNTUyMjkgMiA5IDJIMUMwLjQ0NzcxNSAyIDAgMS41NTIyOCAwIDFaIiBmaWxsPSJ3aGl0ZSIvPgo8L3N2Zz4K');
+      visibility: visible;
     }
   }
+}
 
-  @keyframes checkmarkIn {
-    from {
-      clip-path: inset(16px 0 0 0);
-    }
+input[type='radio'].FormControl-radio {
+  border-radius: var(--primer-borderRadius-full, 100vh);
 
-    to {
-      clip-path: inset(0 0 0 0);
+  &::before {
+    clip-path: circle(0%);
+    mask-image: url('data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAiIGhlaWdodD0iMTAiIHZpZXdCb3g9IjAgMCAxMCAxMCIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPHBhdGggZD0iTTUgOS4zNzVDNy40MTYyMiA5LjM3NSA5LjM3NSA3LjQxNjIyIDkuMzc1IDVDOS4zNzUgMi41ODM3NiA3LjQxNjIyIDAuNjI1IDUgMC42MjVDMi41ODM3NiAwLjYyNSAwLjYyNSAyLjU4Mzc2IDAuNjI1IDVDMC42MjUgNy40MTYyMiAyLjU4Mzc2IDkuMzc1IDUgOS4zNzVaIiBmaWxsPSJ3aGl0ZSIvPgo8L3N2Zz4K'); // checked circle image
+    mask-size: 65%;
+
+    @media screen and (prefers-reduced-motion: no-preference) {
+      animation: radioOut 80ms cubic-bezier(0.65, 0, 0.35, 1) forwards; // slightly snappier animation out
     }
   }
 
-  @keyframes checkmarkOut {
-    from {
-      clip-path: inset(0 0 0 0);
+  &:checked {
+    &::before {
+      @media screen and (prefers-reduced-motion: no-preference) {
+        animation: radioIn 80ms cubic-bezier(0.65, 0, 0.35, 1) forwards 80ms;
+      }
     }
+  }
 
-    to {
-      clip-path: inset(16px 0 0 0);
+  &:indeterminate {
+    &::before {
+      visibility: hidden;
```

Comment:
Where are the rest of the indeterminate styles? I'm not seeing them get rendered in Storybook when I set `indeterminate` to `true`